mcf-browser/src/TimelineExplorer.tsx

```diff
@@ -0,0 +1,390 @@
+        },
+    );
+    locationsPromise.then(
+        (locationOptions) => {
+          const locations = locationOptions.slice(0, INITIAL_NUM_LOCATIONS)
+              .map(
+                  (option) => option.value,
+              );
+          const selectKey = locations.join(',');
+          const locationMapping: LocationMapping = {};
+          locationOptions.map(
+              (obj) => locationMapping[obj.value] = obj.label,
+          );
+          const page = 0;
+          const statVarGroups = Object.values(
+              this.getVariableToSeries(locations),
+          ) as Series[][];
+          this.setState(() => {
+            return {
+              locations,
+              locationOptions,
+              selectKey,
+              locationMapping,
+              statVarGroups,
+              page,
+            };
+          });
+        },
+    );
+  }
+
+  /** Tracks when component is mounted */
+  componentDidMount() {
+    this.updateState();
+  }
+
+  /** Set the location options
+    * @param {TimelineExplorerPropType} prevProps the previous props
+   */
+  componentDidUpdate(prevProps: TimelineExplorerPropType) {
+    if (prevProps.data !== this.props.data) {
+      this.updateState();
+    }
+  }
+
+  /** Returns the series objects which match the current locations filter
+    * @param {string[]} locations a list of locations to filter by
+    * @return {Series[]} series that match the filter
+   */
+  getFilteredData(locations: string[]): Series[] {
+    const cleanedLocations = new Set(locations);
+
+    const filteredData = this.props.data.filter(
+        (series) => cleanedLocations.has(series.observationAbout),
+    );
+
+    return filteredData;
+  }
+
+  /** Processes the data passed in by props and returns the
+    * data grouped by variableMeasured
+    * @param {string[]} locations a list of locations to filter by
+    * @return {Grouping} an object mapping from variableMeasured to an array
+    * of Series with that variableMeasured value
+    */
+  getVariableToSeries(locations: string[]): Grouping {
+    const output: Grouping = {};
+    // Filter by location
+    const data = this.getFilteredData(locations);
+
+    for (const series of data) {
+      const varMeasured = series.variableMeasured ?
+        series.variableMeasured :
+        '';
+      if (!output[varMeasured]) {
+        output[varMeasured] = [];
+      }
+      output[varMeasured].push(series);
+    }
+
+    return output;
+  }
+
+
+  /** Returns the JSX to render a group of related series
+    * @param {Series[]} seriesList a list of series objects with
+    *                              the same varMeasured
+    * @param {LocationMapping} locationMapping a mapping from location dcid to
+    * name
+    * @return {JSX.Element} a details element plotting all of the series
+   */
+  renderSeriesGroup(seriesList: Series[], locationMapping: LocationMapping)
+    : JSX.Element {
+    const varMeasured = seriesList[0].variableMeasured;
```

Comment:
probably want to check that seriesList is not empty before accessing the first element

Reply:
Added!